mosec/coordinator.py

```diff
@@ -67,6 +70,54 @@ def __init__(
             timeout=PROTOCOL_TIMEOUT,
         )
 
+        # optional plugin features - plasma object store as shared memory
+        if shm_path != "":
+            shm_cli = ShmClient(shm_path)
+
+            def get_decoder() -> Callable:
```

Comment:
This not only contains `shm` logic. In fact `shm` logic is very little, just `put` and `get`, but here it needs to combine it with other de/serialization logic, which is obtained from the `Worker`. I feel better to manage here because `Coordinator` is the one who have all these in hand. But we can find a way to remove from `__init__` maybe.

Reply:
I mean the logic can move to an IPC wrapper defined in the `shm` class. It should behave like another layer between the protocol receive/send and the de/serialization.